main/local_modules/frontend_studio/controllers/studio.py

```diff
@@ -1,27 +1,89 @@
 # -*- coding: utf-8 -*-
+import base64
 import logging
 
 from datadog import statsd
-from openerp import http
 from openerp.addons.frontend_base.controllers.base import (Base, QueryURL)
+from openerp.addons.frontend_listing.controllers.listing import Listing
 
+from openerp import http
 from openerp.http import request
 
 _logger = logging.getLogger(__name__)
-_logger.setLevel(logging.DEBUG)
 
 
 class Studio(Base):
     """Representation of the homepage of the website."""
-    studio_url = '/directory/company'
+    partner_url = '/directory/company'
+
+    @http.route(
+        '{0}/<model("res.partner"):partner>/save'.format(partner_url),
+        type='http', auth="public", methods=['POST'], website=True
+    )
+    def save(self, partner,
+             country_id=None,
+             remove_image=False,
+             image_file=None,
+             **kwargs):
+        """Save new data of the partner.
+
+        Data might be converted to be ingest by odoo.
+        For example, list of industries has to be gathered from all the keys
+        starting by industry_ids and converted into an odoo leaf to be ingested
+        into the X2X industry_ids field.
+
+        For the image several options to the user:
+         - a bool (remove_image) that will just remove the current image.
+         - a browse (image_file) that will replace the current image by the
+         newly selected.
+
+        If the remove_image is True, the image_file is ignored.
+
+
+        :param object partner: record of a res.partner.
+        :param int country_id: id of the country to set the partner to.
+        :param bool remove_image: if the current image of the partner should
+            be removed.
+        :param werkzerg.Filestore image_file: instance that represents the
+            new image of the partner.
+        :param dict kwargs: list of additional fields to update.
 
+        :return: request.render
+        """
+        _logger.debug('save')
+        _logger.debug('kwargs: %s', kwargs)
+
+        if country_id:
```

Comment:
The logic should be in `res.partner` and not in the controller.

It will allow also to build tests around the process to be solidify the code.
I also think the parameter in the method of `res.partner` should be defined so it fails if something change in the gui.